python/dgl/distributed/graph_services.py

```diff
@@ -212,13 +239,18 @@ def __getstate__(self):
     def process_request(self, server_state):
         local_g = server_state.graph
         partition_book = server_state.partition_book
+        kv_store = server_state.kv_store
+        if self.prob is not None:
+            prob = kv_store.local_partition[self.prob]
```

Comment:
can this actually work as expected? `process_request` is always called in `DistGraphServer`, so the `kv_store` here is `KVServer`. But `local_partition` is defined in `KVClient`?

Reply:
Nice catch!  I added a unit test.